### DISCUSSION TITLE
Won't launch under new WSL Ubuntu setup

### DISCUSSION THREAD

**Discussion by emendelson** [dosemu2/dosemu2#1986] [General]
Possibly no one cares about this, because it may not matter in the real world, but here's the issue:

I recently set up a new Windows 11 system with Ubuntu Jammy running under WSL. After updating everything, I installed dosemu2, and tried to run it, but got these messages:

```
can't create local /var/run/user/1000/dosemu2 directory

ERROR: cpu exception in dosemu code outside of VM86()!
Floating point exception (Floating-point divide by zero [0x7fae337f2751])
```
Again, there's probably no reason to think about this, but I thought it was worth reporting.

EDIT: I restarted my machine to my old Windows 11 system where dosemu was working perfectly. I updated WSL Ubuntu, and tried to run dosemu, and got a similar error message to the second message above:

```
ERROR: cpu exception in dosemu code outside of VM86()!
Floating point exception (Floating-point divide by zero [0x7ff84efae751])
```
On this older system, the folder `/var/run/user/1000` exists and has a lot of things in it, I don't know why it doesn't exist on my new setup.

**Comment by andrewbird** [comment]
> On this older system, the folder /var/run/user/1000 exists and has a lot of things in it, I don't know why it doesn't exist on my new setup.

`/var/run/user/$uid` is usually created by systemd, I think you installed that as part of your debugging efforts in your other ticket https://github.com/dosemu2/dosemu2/discussions/1892#discussioncomment-5219694

**Comment by stsp** [comment]
> Again, there's probably no reason to think about this, but I thought it was worth reporting.

If you thought its worth reporting, then you
would do it, opening a ticket and following
the instructions within.
Otherwise "there is no reason to think about
this".

**Comment by emendelson** [comment]
Yes, I forgot that the /var/run etc. problem is solved by enabling systemd:

https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/

Will file a bug report on the cpu excpetion.

**Comment by andrewbird** [comment]
> https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/

Interesting, even MS is bowing down to the mighty lord systemd!

**Reply by stsp** [reply]
Just to launch ubuntu?
Not intriguing.
When they integrate it into windows
itself, then that would be a lord. :)

**Comment by andrewbird** [comment]
I think the salient part of the article is giving up PID 1 to systemd.
